Cargo.toml

```diff
@@ -1,33 +1,32 @@
 [package]
 name = "tls-listener"
 description = "wrap incoming Stream of connections in TLS"
-version = "0.3.0"
+version = "0.4.0"
 authors = ["Thayne McCombs <astrothayne@gmail.com>"]
 repository = "https://github.com/tmccombs/tls-listener"
 edition = "2018"
 license = "Apache-2.0"
 
 [features]
 default = ["tokio-net"]
+rustls = ["tokio-rustls"]
+native-tls = ["tokio-native-tls"]
 
 tokio-net = ["tokio/net"]
 hyper-h1 = ["hyper", "hyper/http1"]
 hyper-h2 = ["hyper", "hyper/http2"]
 
 [dependencies]
 futures-util = "0.3.8"
-tokio = "1.0"
-tokio-rustls = "0.23.0"
+hyper = { version = "0.14.1", features = ["server", "tcp"], optional = true }
 pin-project-lite = "0.2.8"
-#tokio-native-tls = "0.3.0"
-
-[dependencies.hyper]
-version = "0.14.1"
-features = ["server"]
-optional = true
+thiserror = "1.0.30"
+tokio = { version = "1.0", features = ["time"] }
```

Comment:
There was a bug here such that if you try to use this crate in another project that doesn't itself also import tokio/time, then it will fail to compile.
You wouldn't have noticed this in the tests because of the weird way Cargo works, where it combines all the feature flag from different targets (i.e. the library + dev dependencies in this case) (something like this https://github.com/rust-lang/cargo/issues/3620 )